pages/book/contracts.mdx

```diff
@@ -0,0 +1,203 @@
+# Contracts
+
+import { Callout } from 'nextra/components'
+
+Contracts in Tact are similar to classes in popular object-oriented languages, except that their instances are deployed on the blockchain and they can't be passed around like [Structs and Messages](/book/structs-and-messages).
+
+## Self-references [#self]
+
+Contracts and [traits](/book/types#traits) have a built-in [identifier](/book/expressions#identifiers) `self{:tact}`, which is used for referring to their fields (persistent state [variables](#variables) and [constants](#variables)) and methods ([internal functions](#internal-functions)):
+
+```tact
+contract Example {
+    // persistent state variables
+    var: Int;
+
+    init() {
+        self.var = 42; // <- referencing the var through self.
+    }
+}
+```
+
+## Structure
+
+Each contract can contain:
```

Comment:
Do we want to mention global functions here?

Reply:
![Internal functions](https://github.com/tact-lang/tact-docs/assets/68142933/24c6c1af-5b46-4eca-a931-fbf238310fcf)

Well, I decided to mention that other kinds of functions exist (global static, extension, etc.) by the end of that page